### PR TITLE
feat: Enhance Proxy Request Handling to Display Actual External URLs

### PR DESCRIPTION
**Pull request by mstarzec386**
This enhancement would improve observability for applications using proxies, enabling us to monitor actual external service calls more accurately.

https://github.com/newrelic/node-newrelic/issues/2688

### CODE REVIEW

File: lib/instrumentation/core/http-outbound.js [hunk 1]
```diff
@@ -58,6 +59,18 @@ function getDefaultHostName(opts) {
   return opts.hostname || opts.host || DEFAULT_HOST
 }
 
+function extractHostPortViaProxy(opts) {
```
Comment (jsumners-nr) [inline lib/instrumentation/core/http-outbound.js:62]: Please document this function.

**Review by jsumners-nr — CHANGES_REQUESTED** [review]
Please add at least one test.

File: lib/instrumentation/core/http-outbound.js (hunk 1 above)
Comment (bizob2828) [reply lib/instrumentation/core/http-outbound.js:62]: I'm also not seeing how this is solving your need.  pathname is coming from `url.parse` which always is a URI and will never have the protocol in the path.

**Comment by mstarzec386** [comment]
Tests added

File: lib/instrumentation/core/http-outbound.js (hunk 1 above)
Comment (mstarzec386) [reply lib/instrumentation/core/http-outbound.js]: Protocol will be present on requests via proxy like:
`proxyRequest.js`
```
require "newrelic";
const axios = require('axios');

axios.get('https://google.com/index.html').then(res => {
  console.info(res.data);
});
```
`HTTPS_PROXY=http://proxy:8080 HTTP_PROXY=http://proxy:8080 NO_PROXY=localhost,.internal node proxyRequest.js`

File: test/integration/instrumentation/http-outbound.tap.js [hunk 2]
```diff
@@ -111,6 +111,43 @@ tap.test('external requests', function (t) {
...
+    })
+    t.teardown(() => proxyServer.close())
+
+    proxyServer.listen(0)
+
+    helper.runInTransaction(agent, function inTransaction() {
+      const opts = {
+        host: 'localhost',
+        port: proxyServer.address().port,
+        path: 'https://www.google.com/proxy/path',
```
Comment (bizob2828) [inline test/integration/instrumentation/http-outbound.tap.js:127]: Overall, nice test. can you just assert the url _could_ get proxied if you didn't terminate the connection?

```suggestion
  t.test('should recognize requests via proxy correctly', function (t) {
    const proxyUrl = 'https://www.google.com/proxy/path'
    const proxyServer = http.createServer(function onRequest(req, res) {
      t.equal(req.url, proxyUrl)
      req.resume()
      res.end('ok')
    })
    t.teardown(() => proxyServer.close())

    proxyServer.listen(0)

    helper.runInTransaction(agent, function inTransaction() {
      const opts = {
        host: 'localhost',
        port: proxyServer.address().port,
        path: proxyUrl
```

**Review by bizob2828 — CHANGES_REQUESTED** [review]
One test suggestion, also @mstarzec386 can you please sign the CLA?

File: test/integration/instrumentation/http-outbound.tap.js (hunk 2 above)
Comment (mstarzec386) [reply test/integration/instrumentation/http-outbound.tap.js]: Hi, assert with proxyUrl added.
CLA signed.

**Review by jsumners-nr — APPROVED** [review]
Looks good to me.

**Review by bizob2828 — APPROVED** [review]

**Comment by jsumners-nr** [comment]
I'll issue a release that includes this on Monday, November 11.

**Comment by jsumners-nr** [comment]
Thank you for the contribution. It has been released as part of v12.7.0.

**Pull request merged**